editors/code/package.json

```diff
@@ -549,6 +549,11 @@
                     "type": "boolean",
                     "default": true
                 },
+                "rust-analyzer.lens.methodReferences": {
+                    "markdownDescription": "Whether to show `Method References` lens. Only applies when `#rust-analyzer.lens.enable#` is set.",
+                    "type": "boolean",
+                    "default": true
```

Comment:
I worry about enabling this by default because of the performance.  This code lens changes big-O of the amount of work we need to do after change. Before it was roughly, O(1) with reference lenses it'll be roughly O(N). Reference search is a fundamentally costly operation.

Moreover, our reference search is super unoptimized at the moment -- at minimum, we need to implement trigram index & fast pruning by syntax tree.

Let's default to false here and elsewhere, but maybe mention somewhere in the manual that you can enable this.

Reply:
Regarding performance: can we put the reference searching into `handle_code_lens_resolve`? Ideally we should be deferring as much code lens work as possible.

Reply:
@kjeremy Sure we can. I just did it :)